src/mapgen/mg_biome.cpp

```diff
@@ -147,18 +149,51 @@ BiomeGenOriginal::BiomeGenOriginal(BiomeManager *biomemgr,
 	// fallback biome when biome generation (which calculates the biomemap IDs)
 	// is disabled.
 	memset(biomemap, 0, sizeof(biome_t) * m_csize.X * m_csize.Z);
+
+	// Calculating the bounding position of each biome so we know when we might switch
+	biomeTransitions = new s16[m_bmgr->getNumObjects() * 2];
+	for (size_t i = 1; i < m_bmgr->getNumObjects(); i++) {
+		Biome *b = (Biome *)m_bmgr->getRaw(i);
+		biomeTransitions[2 * (i - 1)] = b->max_pos.Y;
+		biomeTransitions[2 * (i - 1) + 1] = b->min_pos.Y;
+	}
+
+	// Sorting the biome transition points
+	std::sort(biomeTransitions, biomeTransitions + m_bmgr->getNumObjects() * 2, std::greater<int>());
+
+	// Getting rid of duplicate biome transition points
+	std::vector<s16> t;
+	s16 last = biomeTransitions[0];
```

Comment:
The last two indices are unassigned. You initialize `m_bmgr->getNumObjects() * 2` objects but populate all but the default biome. If there are no biomes registered, `biomeTransitions[0]` returns garbage.

Suggestion: Use a vector + `reserve` from the beginning, sort it and then remove duplicates.